modules/system/classes/extensions/PluginManager.php

```diff
@@ -120,8 +120,389 @@ protected function init(): void
+        }
+
+        $this->renderComponent(Info::class, 'Plugin <fg=yellow>' . $code . '</> installed successfully.');
+
+        // Return an instance of the plugin
+        return $this->findByIdentifier($code);
+    }
+
+    public function isInstalled(ExtensionSource|WinterExtension|string $extension): bool
+    {
+        if (
+            !($code = $this->resolveIdentifier($extension))
+            || $this->versionManager->getCurrentVersion($code) === '0'
+        ) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public function get(WinterExtension|ExtensionSource|string $extension): ?WinterExtension
+    {
+        if (!($code = $this->resolveIdentifier($extension))) {
+            return null;
+        }
+
+        return $this->findByIdentifier($code);
+    }
+
+    /**
+     * Enables the provided plugin using the provided flag (defaults to static::DISABLED_BY_USER)
+     */
+    public function enable(WinterExtension|string $extension, string|bool $flag = self::DISABLED_BY_USER): ?bool
+    {
+        if (!($plugin = $this->get($extension))) {
+            return null;
+        }
+
+        // $flag used to be (bool) $byUser
+        if ($flag === true) {
+            $flag = static::DISABLED_BY_USER;
+        }
+
+        // Unflag the plugin as disabled
+        $this->unflagPlugin($plugin, $flag);
+
+        // Updates the database record for the plugin if required
+        if ($flag === static::DISABLED_BY_USER) {
+            $record = $this->getPluginRecord($plugin);
+            $record->is_disabled = false;
+            $record->save();
+
+            // Clear the cache so that the next request will regenerate the active flags
+            $this->clearFlagCache();
+        }
+
+        // Clear the registration values cache
+        $this->registrationMethodCache = [];
+
+        return true;
+    }
+
+    /**
+     * Disables the provided plugin using the provided flag (defaults to static::DISABLED_BY_USER)
+     */
+    public function disable(WinterExtension|string $extension, string|bool $flag = self::DISABLED_BY_USER): ?bool
+    {
+        if (!($plugin = $this->get($extension))) {
+            return null;
+        }
+
+        // $flag used to be (bool) $byUser
+        if ($flag === true) {
+            $flag = static::DISABLED_BY_USER;
+        }
+
+        // Flag the plugin as disabled
+        $this->flagPlugin($plugin, $flag);
+
+        // Updates the database record for the plugin if required
+        if ($flag === static::DISABLED_BY_USER) {
+            $record = $this->getPluginRecord($plugin);
+            $record->is_disabled = true;
+            $record->save();
+
+            // Clear the cache so that the next request will regenerate the active flags
+            $this->clearFlagCache();
+        }
+
+        // Clear the registration values cache
+        $this->registrationMethodCache = [];
+
+        return true;
+    }
+
+    /**
+     * @throws SystemException
```

Comment:
What throws the SystemException?

Reply:
I think it was something from a previous revision, see: https://github.com/wintercms/winter/pull/1259